src/include/units/quantity_cast.h

```diff
@@ -124,20 +130,20 @@ template<Quantity To, typename D, typename U, typename Rep>
   using c_rep = std::common_type_t<typename To::rep, Rep, intmax_t>;
   using ret_unit = downcast_unit<typename To::dimension, typename To::unit::ratio>;
   using ret = quantity<typename To::dimension, ret_unit, typename To::rep>;
-  using cast = detail::quantity_cast_impl<ret, c_ratio, c_rep, c_ratio::num == 1, c_ratio::den == 1>;
+  using cast = detail::quantity_cast_impl<ret, c_ratio, c_rep, c_ratio::num == 1 && c_ratio::exp == 0, c_ratio::den == 1 && c_ratio::exp == 0>;
```

Comment:
This dispatch is mandated by the C++ standard itself to help with optimizations and provide (double-like performance). It might be needed to provide even more dispatch on `bool ExpIsZero = false`, but we can probably leave it as a TODO and should definitely check the codegen in Compiler Explorer.

Reply:
I don't fully get that. My change, as you could probably tell, was just "to make it work". 

But I take your word for it ;-)